library/src/lapack/roclapack_getrf.hpp

```diff
@@ -437,38 +424,53 @@ rocblas_status rocsolver_getrf_template(rocblas_handle handle,
-    threads = dim3((PIVOT ? BLOCKSIZE : 1), 1, 1);
+    threads = dim3((pivot ? BLOCKSIZE : 1), 1, 1);
 
-    // size of outter blocks
-    blk = getrf_get_blksize<ISBATCHED, PIVOT>(dim);
+    // size of outer blocks
+    blk = getrf_get_blksize<ISBATCHED>(dim, pivot);
 
     if(blk == 1)
-        return rocsolver_getf2_template<ISBATCHED, PIVOT, T>(handle, m, n, A, shiftA, lda, strideA,
-                                                             ipiv, shiftP, strideP, info, batch_count,
-                                                             scalars, pivotval, pivotidx);
+        return rocsolver_getf2_template<ISBATCHED, T>(handle, m, n, A, shiftA, lda, strideA, ipiv,
+                                                      shiftP, strideP, info, batch_count, scalars,
+                                                      pivotval, pivotidx);
 
     // MAIN LOOP =====>
     for(rocblas_int j = 0; j < dim; j += blk)
     {
         jb = min(dim - j, blk); // number of columns/pivots in the block
-        blk1 = getrf_get_innerBlkSize<ISBATCHED, PIVOT>(m - j, jb); // size of inner blocks
+        blk1 = getrf_get_innerBlkSize<ISBATCHED>(m - j, jb, pivot); // size of inner blocks
 
         // LOOP FACTORIZING INNER BLOCKS =====>
         for(rocblas_int k = 0; k < jb; k += blk1)
         {
             jb1 = min(jb - k, blk1); // number of columns/pivots in the inner block
 
-            // factorize inner block panel
-            rocsolver_getf2_template<ISBATCHED, PIVOT, T>(
-                handle, m - j - k, jb1, A, shiftA + idx2D(j + k, j + k, lda), lda, strideA, iipiv,
-                0, jb1, iinfo, batch_count, scalars, pivotval, pivotidx);
-
-            // adjust pivots, swap rows and check singularity
-            hipLaunchKernelGGL((getrf_check_singularity<PIVOT, T>), grid, threads, 0, stream, n,
-                               j + k, jb1, A, shiftA, lda, strideA, ipiv, shiftP, strideP, iipiv,
-                               iinfo, info);
+            if(pivot)
+            {
+                // factorize inner block panel
+                rocsolver_getf2_template<ISBATCHED, T>(
+                    handle, m - j - k, jb1, A, shiftA + idx2D(j + k, j + k, lda), lda, strideA,
+                    iipiv, 0, jb1, iinfo, batch_count, scalars, pivotval, pivotidx);
+
+                // adjust pivots, swap rows and check singularity
+                hipLaunchKernelGGL(getrf_check_singularity<T>, grid, threads, 0, stream, n, j + k,
+                                   jb1, A, shiftA, lda, strideA, ipiv, shiftP, strideP, iipiv,
+                                   iinfo, info);
+            }
+            else
+            {
+                // factorize inner block panel
+                rocsolver_getf2_template<ISBATCHED, T>(
+                    handle, m - j - k, jb1, A, shiftA + idx2D(j + k, j + k, lda), lda, strideA,
+                    nullptr, 0, jb1, iinfo, batch_count, scalars, pivotval, pivotidx);
```

Comment:
is not `iipiv = nullptr` a necessary and sufficient condition for `getf2_template` to avoid pivoting? I think you can get this call out of the if statement. 

(also, if you pass `pivot` to the templates, you can have the call out of the if statement regardless of the value of `iipiv`. After all, `pivot` should be set to the right value right at the beginning: at the C API wrapper level).